tsl/src/bgw_policy/job_api.c

```diff
@@ -436,6 +443,13 @@ job_alter(PG_FUNCTION_ARGS)
 		job->fd.initial_start = initial_start;
 	}
 
+	if (!PG_ARGISNULL(13))
+	{
+		char app_name[NAMEDATALEN];
+		snprintf(app_name, NAMEDATALEN, "%s [%d]", NameStr(*PG_GETARG_NAME(13)), job_id);
```

Comment:
Job ID is already available in the table, so not sure if it is necessary to add it here as well.